ray_provider/decorators/ray_decorators.py

```diff
@@ -64,9 +73,111 @@ def sum_cols(df: pd.DataFrame) -> pd.DataFrame:
     @functools.wraps(python_callable)
     def wrapper(f):
 
+        def on_execute_callback(context):
+            pass
+
+        @provide_session
+        def on_retry_callback(context, session=None):
+
+            # Connect to 'airflow' namespace to access scoped actors
+            if not ray.util.client.ray.is_connected():
+                ray.util.connect('192.168.1.70:10001', namespace='airflow')
+
+            # Executes GCS download and unpickling from within Ray
+            def _load_gcs_within_ray(object_name, bucket_name='astro-ray'):
+
+                from google.cloud import storage
+                import os
+                import pickle
+
+                # GCS conn points creds to path on Ray server
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = '/Users/p/code/gcs/astronomer-ray-demo-87cd7cd7e58f.json'
+
+                # Read target value from GoogleCloudStorage
+                bucket = storage.Client().bucket(bucket_name)
+                obj_from_gcs = bucket.blob(
```

Comment:
@richardliaw we'd like some feedback on this part. Seems like downloading as a string might limit size. Is there a "best practice" that Ray does for object storage/retrieval?